lib/utils/uniqueID.js

```diff
@@ -0,0 +1,9 @@
+var index = 0
+
+// This is of course not using UUIDs, but should be totally sufficient
+// for our use case
+function uniqueID () {
+  return 'faux-dom-' + (index++)
+}
```

Comment:
> I build everything stateless unless state is required because it leads to cleaner and less buggy code

State *is* required in this case. I'm totally for minimising state, but in this case you're just handing the task of managing that state to your users.

The point is, this is not perfect when we're creating elements on the fly, but that's when the users should help out and provide keys themselves -- which then doesn't make a difference to what you're proposing. React's heuristic 1-1 diffing of lists without keys is also really sub optimal.

But when we're creating elements beforehand, the user doesn't have to do anything. This is a *huge* step forward.

I'd focus on how we can either try to remove our keys when we're creating elements on the fly, or how we can output nice warnings in that case, or both.